### DISCUSSION TITLE
more ways to add large powershell script

### DISCUSSION THREAD

**Discussion by Omar7saeed** [Tyrrrz/CliWrap#116] [Q&A] [answered]
hello again :)

sometimes when I convert script to -encodedcommand  shape to add it like this 

`.WithArguments(new[] {"-NoProfile", "-EncodedCommand", "VwByAGkAdAlAC0A..ETC"})`

if the script size is large... the PowerShell will not run the encoded command and give an Error
are there any other ways?

I tried to add the original script within  " .., ", but I think i can't ...so many c# errors

**Comment by Tyrrrz** [comment]
Store the script in a file?

**Reply by Omar7saeed** [reply]
hi

I was trying to add the ps1 as a command 
but if I must embed the script itself in my situation .. no problem 

can u help me call it from the resources please ? ... like the last time when u typed me a code
I'm sorry dear for being so noob .. but I just started programming a few days ago.

thank u Alexey

**Reply by Tyrrrz** [reply]
I don't think you can call it from resources directly, but you can copy it to a temporary file, then use `powershell -File path/to/script/ps1`. But also `EncodedCommand` should work unless I guess your command is really huge (I'm not sure what's the limit is).

**Reply by Omar7saeed** [reply]
yes, `EncodedCommand` working like a charm, but I exceed the limit...
copy it to temp is a great suggest 

I have an idea dear ... I will try it first... if I failed I will do your suggest 
in my script situation, I can spilt the script into 2 parts ... then execute it via `EncodedCommand`

but there is a part that must run first ... is there any way can confirm that?
delay as example or command 

thank u brother You have helped me a lot, you shortened the long road for me

**Reply by Omar7saeed** [reply]
@Tyrrrz

**Reply by Tyrrrz** [reply]
If the execution completes, that means it completed.

**Comment by Omar7saeed** [comment, marked as answer]
fixed by spilt the script into 2 parts ... then execute it via EncodedCommand.
